app/models/barcode.rb

```diff
@@ -11,9 +11,18 @@ class Barcode < ApplicationRecord
   require 'sanger_barcode_format/legacy_methods'
   extend SBCF::LegacyMethods
 
+  # This association will have to be removed in a subsequent story
   belongs_to :asset, optional: false, class_name: 'Labware'
```

Comment:
With this declaration staying, we have a duplicate declaration for the same association. One with belongs_to: asset and another with belongs_to :labware . This doesn't look right to me. What happens when you try to save it? Do you save it twice? I think,  belongs_to :asset, optional: false, class_name: 'Labware' could be removed. I see in the comments that it is causing validation failure. What was failing?

Reply:
Hi @yoldas, it doesn't really save twice; check [`spec/models/barcode_spec.rb`](https://github.com/sanger/sequencescape/blob/95f1fc2cd612a6db675e99ac42cd837570d6977c/spec/models/barcode_spec.rb#L98-L116).

I didn't **remove** `asset` because this story's goal was primarily to add an alias. If we remove that association, validations do fail. Here are some of the failures:

```
...
* library_aliquot - Could not find the inverse association for barcodes (:asset in Barcode) (ActiveRecord::InverseOfAssociationNotFoundError)
* phi_x_aliquot - Could not find the inverse association for barcodes (:asset in Barcode) (ActiveRecord::InverseOfAssociationNotFoundError)
* barcode - undefined method `classify' for nil:NilClass (NoMethodError)
 ....
```

It speaks of an inverse association for barcodes. I suppose that a lot of validations failures occur because of `inverse_of` in [`app/models/labware.rb:22`](https://github.com/sanger/sequencescape/blob/d9b1df66771ba2124209f9411ba2493cecd458a1/app/models/labware.rb#L22). But I could not get it to rename to `labware` as I could not get the factories fixed because I couldn't locate where it was referred in the factories. I didn't go too deep into it because I think the purpose of the story is to change code references where the `barcode` model invokes `asset` association.

If you've got ideas on some updates, maybe we both can pair it?

Reply:
Aren't those because of the asset line in barcode factory? spec/factories/barcode_factories.rb

```
  factory :barcode, aliases: [:external] do
    asset factory: %i[labware]
```

Reply:
If we change the inverse_of to labware in Labware mode and change the factory as well. Doesn't that fix them?
has_many :barcodes, foreign_key: :asset_id, inverse_of: :asset, dependent: :destroy

Reply:
It _was_ renamed to labware. See [`spec/factories/barcode_factories.rb`](https://github.com/sanger/sequencescape/blob/95f1fc2cd612a6db675e99ac42cd837570d6977c/spec/factories/barcode_factories.rb#L7).